server/athenian/api/controllers/goal_controller.py

```diff
@@ -137,7 +137,7 @@ async def create_goal_template(request: AthenianWebRequest, body: dict) -> web.R
     await get_user_account_status_from_request(request, create_request.account)
 
     repositories = await parse_request_repositories(
-        create_request.repositories, request, create_request.account,
+        create_request.repositories, LazyPrefixerProxy(request, create_request.account),
```

Comment:
Also, if we remove the proxy, we will follow the same conventions that exist in the rest of the codebase, so it should help future programmers

Reply:
Looking at the example, without proxy you can:

1 - always build prefixer, you do useless work; imagine cond_a or cond_b is True 10% of the time:
```
async def controller():
   prefixer = Prefixer()
   awaith gather(a(prefixer), b(prefixer))
async def a(prefixer):
   if  cond_a:
        prefixer.do_something()
async def b(prefixer):
   if  cond_b:
          prefixer.do_something()
```

2 - check conditions also in main function, propagate an optional prefixer, code gets complex, 
you repeat condition check, controller() has to mess with logic it shouldn't know, a() and b() has to mess with 
the assumption that the resource they need is there when they need it

```
async def controller():
   if cond_a or cond_b:
        maybe_prefixer = Prefixer()
   else:
        maybe_prefixer = None
   awaith gather(a(maybe_prefixer), b(maybe_prefixer))
async def a(prefixer):
   if  cond_a:
         assert prefixer is not None
        prefixer.do_something()
async def b(prefixer):
   if  cond_b:
          assert prefixer is not None
          prefixer.do_something()
```

3 - build prefixer twice, this is still better than 1 -  if cond_a or cond_b is False 10% of the time
```
async def controller():
   awaith gather(a(), b())
async def a():
   if  cond_a:
        prefixer = Prefixer()
        prefixer.do_something()
async def b():
   if  cond_b:
          prefixer = Prefixer()
          prefixer.do_something()
```

Reply:
So I understand you suggest going for 1 - and trust memcached